Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+RequestQueue.swift

```diff
@@ -0,0 +1,115 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+extension HTTPConnectionPool {
+    /// A struct to store all queued requests.
+    struct RequestQueue {
+        private var generalPurposeQueue: CircularBuffer<Request>
+        private var eventLoopQueues: [EventLoopID: CircularBuffer<Request>]
+
+        init() {
+            self.generalPurposeQueue = CircularBuffer(initialCapacity: 32)
+            self.eventLoopQueues = [:]
+        }
+
+        var count: Int {
+            self.generalPurposeQueue.count
+        }
+
+        var isEmpty: Bool {
+            self.generalPurposeQueue.isEmpty
+        }
```

Comment:
This is surprising behaviour I think -- I would expect the `count` to return the total count and `isEmpty` to only return true if both queues are empty.